frame/contracts/src/exec.rs

```diff
@@ -510,9 +502,35 @@ where
 		Ok((dest, output))
 	}
 
-	fn new_call_context<'b>(&'b mut self, caller: T::AccountId, value: BalanceOf<T>)
-		-> CallContext<'b, 'a, T, V, L>
-	{
+	pub fn terminate(
+		&mut self,
+		beneficiary: &T::AccountId,
+		gas_meter: &mut GasMeter<T>,
+	) -> Result<(), DispatchError> {
+		let self_id = self.self_account.clone();
+		let value = self.overlay.get_balance(&self_id);
+		if let Some(parent) = self.parent {
```

Comment:
what is the parent of an account here? (totally random question)

Reply:
It is the parent `ExecutionContext`. So basically the calling contract. We check if this contract is already on the call stack (re-entrance)  and if so deny the deletion.

Reply:
Might be a good sign to consider to rename this field to `parent_ctx` or smth.

Reply:
#5270